src/charm.py

```diff
@@ -232,6 +232,13 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent) -> None:
 
         # Update the list of the cluster members in the replicas to make them know each other.
         # Update the cluster members in this unit (updating patroni configuration).
+        container = self.unit.get_container("postgresql")
+        if not container.can_connect():
+            logger.debug(
+                "Deferring on_peer_relation_changed: Waiting for container to become available"
+            )
+            event.defer()
+            return
```

Comment:
I'm not sure if we should defer here. If the workload is not fully up, the configuration should be generated once pebble is ready. Thoughts?

Reply:
I think it's ok to just return here, and leave to pebble ready event to update the config. If, later, the config needs to be updated again, this or some other hooks will be fired again (to call update_config).